rich/pretty.py

```diff
@@ -500,6 +515,50 @@ def iter_rich_args(rich_args) -> Iterable[Union[Any, Tuple[str, Any]]]:
                 node = Node(
                     value_repr=f"{obj.__class__.__name__}()", children=[], last=root
                 )
+        elif _is_attr_object(obj):
+            children = []
+            append = children.append
+
+            attr_fields = _get_attr_fields(obj)
+            if attr_fields:
+                node = Node(
+                    open_brace=f"{obj.__class__.__name__}(",
+                    close_brace=")",
+                    children=children,
+                    last=root,
+                )
+
+                def iter_attrs() -> Iterable[
+                    Tuple[str, Any, Optional[Callable[[Any], str]]]
+                ]:
+                    """Iterate over attr fields and values."""
+                    for attr in attr_fields:
+                        if attr.repr:
+                            try:
+                                value = getattr(obj, attr.name)
+                            except Exception as error:
+                                # Can happen, albeit rarely
+                                yield (attr.name, error, None)
+                            else:
+                                yield (
+                                    attr.name,
+                                    value,
+                                    attr.repr if callable(attr.repr) else None,
```

Comment:
What exactly does this logic do? `repr` being a callable doesn't mean that you have to call it yourself, it's used by the object's `__repr__`. Or do you use it as a marker, that _you_ shouldn't try to render it?

Reply:
Rich will only call `repr` on atomic values. For containers, Rich will take over the rendering of the repr so that it can know how to expand it on to several lines if required.

This logic is so that the custom `repr` will be called if supplied (L551), otherwise it will recurse in to the value (L553), potentially expanding lists and dicts etc. on to several lines.

Does that make sense, or have I misunderstood the purpose of Attribute.repr ?

Reply:
No I think you've got it. A truish repr for an attribute means that `attrs` include it in `__repr__` and if it's a callable, it will use that callable to generate said repr instead of calling `repr()` on the value: https://github.com/python-attrs/attrs/blob/8ae2d6f1c3c067c941fd9d212e52972c7d3382cc/src/attr/_make.py#L1855-L1858